src/sas/sascalc/file_converter/nxcansas_writer.py

```diff
@@ -286,16 +286,20 @@ def _write_h5_vector(entry, vector, names=['x_position', 'y_position'],
                                        data=trans.transmission_deviation)
             trans_entry.create_dataset('lambda', data=trans.wavelength)
 
-        note_entry = sasentry.create_group('sasnote'.format(i))
-        note_entry.attrs['canSAS_class'] = 'SASnote'
-        notes = None
-        if len(data_info.notes) > 1:
+        if np.isscalar(data_info.notes) and data_info.notes:
+            # Handle scalars that aren't empty arrays, None, '', etc.
+            notes = [np.string_(data_info.notes)]
+        elif len(data_info.notes) > 1:
+            # Handle iterables that aren't empty
             notes = [np.string_(n) for n in data_info.notes]
-            notes = np.array(notes)
-        elif data_info.notes != []:
-            notes = _h5_string(data_info.notes[0])
+        else:
+            # Notes are required in NXcanSAS format
+            notes = [np.string_('')]
```

Comment:
`np.string_` is [almost undocumented](https://numpy.org/devdocs/reference/routines.char.html).  It has the same docstring as `bytes` but it is not equal to `bytes` nor is it a subclass of `bytes`.  I suspect it is a holdover from 2-3 conversion, and you could `bytes` directly, or in this case, `b''`.